packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1292,6 +1302,41 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     };
   }
 
+  private configureSnapStart(props: FunctionProps): CfnFunction.SnapStartProperty | undefined {
+    // return/exit if no snapStart included
+    if (!props.snapStart) {
+      return undefined;
+    }
+
+    // SnapStart does not support arm64 architecture, Amazon Elastic File System (Amazon EFS), or ephemeral storage greater than 512 MB.
+    // SnapStart doesn't support provisioned concurrency either, but that's configured at the version level,
+    // so it can't be checked at function set up time
+    // SnapStart supports the Java 11 and Java 17 (java11 and java17) managed runtimes.
+    // See https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html
+    if (props.snapStart != SnapStartConf.ON_PUBLISHED_VERSIONS) {
```

Comment:
Last thing and then we can approve this. Can we remove this? The only way a user can use `SnapStartConf` is as `SnapStartConf.ON_PUBLISHED_VERSIONS`